site/js/modules/webui/server.js

```diff
@@ -317,6 +319,135 @@ function callEndpoint(targetEndpoint, inputFields, context, container) {
+            `;
+
+            container.querySelector(".endpoints input").addEventListener("input", function(event) {
+                container.querySelectorAll(".endpoint").forEach(function(endpointDiv) {
+                    let endpoint = endpointDiv.getAttribute("data-endpoint");
+                    if (endpoint.toLowerCase().includes(event.target.value.toLowerCase())) {
+                        endpointDiv.classList.remove("hidden");
+                    } else {
+                        endpointDiv.classList.add("hidden");
+                    }
+                });
+            });
+
+            container.querySelector(".types input").addEventListener("input", function(event) {
+                container.querySelectorAll(".type").forEach(function(typeDiv) {
+                    let type = typeDiv.getAttribute("data-type");
+                    if (type.toLowerCase().includes(event.target.value.toLowerCase())) {
+                        typeDiv.classList.remove("hidden");
+                    } else {
+                        typeDiv.classList.add("hidden");
+                    }
+                });
+            });
+        })
+        .catch(function (message) {
+            console.error(message);
+            container.innerHTML = Render.autograderError(message);
+        })
+    ;
+}
+
+function displayEndpoints(endpointData) {
+    let endpoints = [];
+
+    Object.entries(endpointData).forEach(function([endpoint, data]) {
+        let args = {
+            [Routing.PARAM_TARGET_ENDPOINT]: endpoint,
+        };
+
+        let inputTypes = Render.tableFromDictionaries({
+            head: [["name", "Name"], ["type", "Type"]],
+            body: data.input,
+        });
+
+        let outputTypes = Render.tableFromDictionaries({
+            head: [["name", "Name"], ["type", "Type"]],
+            body: data.output,
+        });
+
+        endpoints.push(`
+            <div class='endpoint' data-endpoint='${endpoint}'>
+                <a href="${Routing.formHashPath(Routing.PATH_SERVER_CALL_API, args)}">
+                    <h3>${endpoint}</h3>
+                </a>
+                <p>${data.description}</p>
+                <div class="details">
+                    <div>
+                        <h4>Input</h4>
+                        ${inputTypes}
+                    </div>
+                    <div>
+                        <h4>Output</h4>
+                        ${outputTypes}
+                    </div>
+                </div>
+            </div>
+        `);
+    });
+
+    return endpoints.join("\n");
+}
+
+function displayTypes(typeData) {
+    let types = [];
+
+    Object.entries(typeData).forEach(function([type, data]) {
+        let fieldData = "";
+        if (data.fields) {
+            let fieldTypes = Render.tableFromDictionaries({
+                head: [["name", "Name"], ["type", "Type"]],
+                body: data.fields,
+            });
+
+            fieldData += `
+            	<div class="details">
```

Comment:
"details" and "type" just seems a little too general for classes.
Of course, you are querying for it properly by adding the `.api-docs` context,
but I would still like them to be a little more specific.

(Note that there is another comment here because you don't add the context for a `.type` query.)